### PR TITLE
BUG: add int->int loop for trunc

### LINKED ISSUE #19464: `np.trunc` is inconsistent with array-api
As per `array-api`,
```
The returned array must have the same data type as x .
```

However,
```python
>>> np.__version__
'1.21.0'
>>> np.trunc(2).dtype
dtype('float64')
>>> np.trunc(np.ones(2, dtype=np.int32)).dtype
dtype('float64')
```

Ref: https://data-apis.org/array-api/latest/API_specification/elementwise_functions.html#trunc-x

### PR DESCRIPTION
**Pull request by kshitij12345**
Fixes: #19464 

Not entirely sure if this change is automatically picked up by `test_ufunc.py`. Would be great to know if this needs more explicit test.

Thanks!

**Note**: BC-Breaking

cc: @seberg

### CODE REVIEW

**Comment by seberg** [comment]
Thanks, I think this should have a brief release note at least (see `doc/release/upcoming_changes` there is a `README` there.  And since I am not sure how conservative we should be here: could you ping the mailing list about the change?  We _could_ give this a proper `FutureWarning`, although it would be a bit annoying...

Also needs at least some basic tests (although it looks like the code actually gets executed somewhere already).

**Comment by kshitij12345** [comment]
> could you ping the mailing list about the change?

Sure. Will do that.

Thanks!!

**Comment by kshitij12345** [comment]
Link to the discussion: http://numpy-discussion.10968.n7.nabble.com/np-trunc-is-inconsistent-with-array-api-td49360.html

**Comment by seberg** [comment]
This is confusing me, there seem to be some emails there that should be there and were posted to the actual mailing list?  But then there are also definitely mails missing and some additional mails (like yours).

But does nabble just keep its own stuff around and it is not connected to the actual mailing list, which is [hosted on python.org](https://mail.python.org/mailman/listinfo/numpy-discussion)?

**Comment by mattip** [comment]
@kshitij12345 I am not sure how you sent your mail, but I think it did not actually go out to the mailing list even though it somehow shows up in nabble. Could you (perhaps subscribe to https://mail.python.org/mailman/listinfo/numpy-discussion and) re-send the mail to numpy-discussion@python.org ?

**Comment by kshitij12345** [comment]
Ah. Sure. Sorry for the confusion.

Thanks @seberg @mattip

**Comment by mattip** [comment]
In any case this looks good. The other ufuncs in that family: `rint`, `fix`, `ceil`, `floor` have the same problem.

**Comment by mattip** [comment]
@kshitij12345 did you get a chance to look at the other ufuncs and/or ping the mailing list about this?

**Comment by kshitij12345** [comment]
@mattip Ah sorry. I missed doing that. Will do both of them in a day or two. Thanks for the ping :)

File: numpy/core/tests/test_ufunc.py [hunk 1]
```diff
@@ -612,6 +612,13 @@ def test_true_divide(self):
         assert_(res == 0.0)
         assert_(res.dtype.name == 'float64')
 
+    def test_trunc_expected_dtypes(self):
+        # Supported dtypes by trunc.
+        for tc in 'bBhHiIlLqQefdgO':
```
Comment (seberg) [inline numpy/core/tests/test_ufunc.py:617]: What about boolean `?`?  (What should that actually do?)

Other than that, the only question is whether we should not aim for fixing this for the whole class of truncating functions?

Comment (kshitij12345) [reply numpy/core/tests/test_ufunc.py:617]: Ah. Right now, `bool` get promoted which we don't want. Will add a loop entry for `bool`.

As for others, is it ok to do a follow-up PR.

Apologies for the delayed response. (Will be more active henceforth :))

Thanks!

**Pull request closed without merging**